core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -257,7 +257,7 @@ object DumpLogSegments {
     private def parseGroupMetadata(groupMetadataKey: GroupMetadataKey, payload: ByteBuffer) = {
       val groupId = groupMetadataKey.key
       val group = GroupMetadataManager.readGroupMessageValue(groupId, payload)
-      val protocolType = group.protocolType
+      val protocolType = group.protocolType.getOrElse("NONE")
```

Comment:
Not a big deal for this code and it was already merged, so fine to stay as is.

My reasoning is that you lose a bit of type safety by converting from `Option[String]` to a `String` sentinel value and then doing an equality comparison on it. We should never have a protocol with name `""`, so unlikely to matter here, but I generally prefer to keep the types for as long as possible as it helps avoid bugs as the code evolves.

As an aside, this regression would never have happened if Scala had a type safe equals by default (a number of libraries have a `===` which is a type safe version of `==` as a workaround).